scripts/commands/addcurrency.lua

```diff
@@ -9,33 +9,125 @@ cmdprops =
     parameters = "sis"
 };
 
-function onTrigger(player,currency,amount,target)
-    local targ = nil
+function error(player, msg)
+    player:PrintToPlayer(msg);
+    player:PrintToPlayer("@addcurrency <currency type> <amount> {player}");
+end;
 
-    if (currency == nil or amount == nil) then
-        player:PrintToPlayer( "You must enter a valid currency and amount." );
-        player:PrintToPlayer( "@addcurrency <currency type> <amount> <target player>" );
-        return
-    end
+function onTrigger(player,currency,amount,target)
+    local currencies = {
+        sandoria_cp=true,
```

Comment:
What exactly happened when you tried it?

The entire reason these are strings is to avoid having to lua side define them at all (technically bad practice, but super convenient). Last time I tried trying to pass non-existent currency to this error'd out like it was supposed to in the core.